lib/kiev/shoryuken/request_id.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "securerandom"
+
+module Kiev
+  module Shoryuken
+    class RequestId
+      REQUEST_ID = "request_id"
+      REQUEST_DEPTH = "request_depth"
+      TREE_PATH = "tree_path"
+
+      def call(_worker, _queue, message, _body)
+        Kiev::RequestStore.store[:request_id] = request_id(message)
+        Kiev::RequestStore.store[:request_depth] = request_depth(message)
+        Kiev::RequestStore.store[:tree_path] = tree_path(message)
+        yield
+      end
+
+      private
+
+      def request_id(message)
+        message_attribute_value(message, REQUEST_ID) || SecureRandom.uuid
```

Comment:
Also it'd be nice to have `SecureRandom.uuid` to be overridable here, at least with monkey-patching. Different apps may have different requirements for their ids. Since it's open-source gem this must be achievable.

Reply:
I'm not making that call.
But if it were up to me, I would impose a random UUID as a convention anyway.